src/stream_base.h

```diff
@@ -29,7 +29,10 @@ using JSMethodFunction = void(const v8::FunctionCallbackInfo<v8::Value>& args);
 
 class StreamReq {
  public:
-  static constexpr int kStreamReqField = 1;
+  enum InternalFields {
+    kStreamReqField = BaseObject::kInternalFieldCount,
```

Comment:
This doesn't look right. `StreamReq` doesn't derive from `BaseObject`.

The way `StreamReq::kStreamReqField` is currently used (not in this PR) looks very dubious to me:
```c++
sw->InstanceTemplate()->SetInternalFieldCount(StreamReq::kStreamReqField + 1 + 3);
```
I mean, what?

Reply:
Your guess is as good as mine on this one. I was going to revisit this one once I figured it out :-)

Reply:
Ah, ok, `StreamReq` does not derive from `BaseObject` but a few subclasses down in the hierarchy do... specifically `SimpleShutdownWrap` and `SimpleWriteWrap`. These are currently the only things that derive from `ShutdownWrap` and `WriteWrap` (respectively), which derive from `StreamReq`. I'm still not 100% sure what's up with the internal field count value, however.

Reply:
For both `StreamReq` and `StreamResource`, I've added their own `kSlot` definitions to their enums but added comments to explain that those mirror the `BaseObject::InternalFields::kSlot` definition and why. This can be revisited later if this part of the code is refactored further.